include/rocm_smi/rocm_smi.h

```diff
@@ -239,6 +238,24 @@ typedef struct {
     uint64_t frequency[RSMI_MAX_NUM_FREQUENCIES];
 } rsmi_frequencies;
 
+/**
+ * @brief This structure holds information about the possible PCIe
+ * bandwidths. Specifically, the possible transfer rates and their
+ * associated numbers of lanes are stored here.
+ */
+typedef struct {
+    /**
+     * Transfer rates (T/s) that are possible
+     */
+    rsmi_frequencies transfer_rate;
+
+    /**
+     * List of lanes for corresponding transfer rate.
+     * Only the first num_supported bandwidths are valid.
+     */
+    uint32_t lanes[RSMI_MAX_NUM_FREQUENCIES];
```

Comment:
Is RSMI_MAX_NUM_FREQUENCIES fixed by something external to ROCM-SMI?  If this changes how will you handle backward compatibility?

Reply:
It's set to 32 just because it matches the size of a 32-bit bit mask that determines which of up to 32 different frequencies are allowed. Right now, I think there are around 5-7 frequencies to choose from. I'm not sure of what the most elegant way of ensuring extensibility for the possibility that it will grow past 32.  We need to add versioning of the api for this. In future versions, we could add flexibility to some of these structures by adding some size or versioning info to the first bytes of the structure.